connections/README.md

```diff
@@ -325,6 +326,17 @@ Resource allocation, measurement and enforcement policies are all an active area
 of discussion in the libp2p community, and implementations are free to develop
 whatever prioritization system makes sense.
 
+#### Supported protocols
+
+A libp2p node SHOULD scope its set of supported protocols to the underlying
+physical connection to a peer. It MAY only support a protocol based on properties
+of a physical connection to e.g. limit the use of bandwidth-heavy protocols over
+a relayed or metered connection. A libp2p node MAY offer different sets of protocols
+to different peers. It MAY revoke or add the support for a protocol at any time,
```

Comment:
I am not sure. This section of the specs is pretty ambivalent about protocols. Do we really want to introduce this here?

I like the idea but I think it would be better added to the identify spec rather than here.